src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlDependency.cs

```diff
@@ -241,29 +241,33 @@ private static void InvokeCallback(object eventContextPair)
         // END EventContextPair private class.
         // ----------------------------------------
 
-        // ----------------------------------------
-        // Private class for restricting allowed types from deserialization.
-        // ----------------------------------------
 
-        private class SqlDependencyProcessDispatcherSerializationBinder : SerializationBinder
+        //-----------------------------------------------
+        // Private Class to add ObjRef as DataContract
+        //-----------------------------------------------
+        [SecurityPermission(SecurityAction.Assert, Flags = SecurityPermissionFlag.RemotingConfiguration)]
+        [DataContract]
+        private class SqlClientObjRef : IExtensibleDataObject
         {
-            public override Type BindToType(string assemblyName, string typeName)
+            [DataMember]
+            private static ObjRef SqlObjRef;
+
+            public SqlClientObjRef(SqlDependencyProcessDispatcher dispatcher)
             {
-                // Deserializing an unexpected type can inject objects with malicious side effects.
-                // If the type is unexpected, throw an exception to stop deserialization.
-                if (typeName == nameof(SqlDependencyProcessDispatcher))
-                {
-                    return typeof(SqlDependencyProcessDispatcher);
-                }
-                else
-                {
-                    throw new ArgumentException("Unexpected type", nameof(typeName));
-                }
+                SqlObjRef = RemotingServices.Marshal(dispatcher);
+            }
+
+            private ExtensionDataObject _extensionData_Value;
+
+            public ExtensionDataObject ExtensionData
```

Comment:
Any purpose we keep this property? I don't see this being used.

Reply:
that comes with IExtensibleDataObject. That interface implements that part. I have changed it to use Auto property I believe. 
you can read more [here ](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.serialization.iextensibledataobject?view=net-5.0#remarks). I will test to see if  we can use the class without the interface though.

Reply:
Since this is a private class and has an internal usage, if we had more data members in the future, we could decide about the versioning solution later. So, I believe you can remove that interface.